### PR TITLE
Product detail basic functionality

### PR DESCRIPTION
**Pull request by slargman**
Adds the basic functionality for the product detail section of the overview. No styling has been added yet, just the display of product information. Also sets up the data flow in the parent Overview component.

### CODE REVIEW

File: client/src/components/Overview/Overview.jsx [hunk 1]
```diff
@@ -1,14 +1,64 @@
...
+          axios.get(`/products/${productId}/styles`),
+        ]);
+
+        const sortedStyles = stylesResponse.data.results.sort((style1, style2) => (
+          style1.styled_id - style2.styled_id
+        ));
+
+        setProduct(productResponse.data);
+        setStyles(sortedStyles);
+        setSelectedStyleId(sortedStyles.find(style => style['default?']).style_id);
```
Comment (zman811) [inline client/src/components/Overview/Overview.jsx:28]: I believe for the style guide it wants you to have parens even if its only 1 arg

File: client/src/components/Overview/ProductInformation.jsx [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
 import StyleSelector from 'Overview/StyleSelector.jsx';
 import AddToCart from 'Overview/AddToCart.jsx';
 
-function ProductInformation() {
+/**
+ * Shows general product information
+ */
+function ProductInformation({ product, styles, selectedStyleId, handleStyleSelect }) {
+  // TODO: how to avoid optional chaining?
+  const selectedStyle = styles?.find(style => style.style_id === selectedStyleId);
```
Comment (zman811) [inline client/src/components/Overview/ProductInformation.jsx:11]: same thing here though I think you are planning to change this anyway

**Review by zman811 — APPROVED** [review]
Looks great though, hard for me to find anything to change haha

File: client/src/components/Overview/Overview.jsx (hunk 1 above)
Comment (slargman) [reply client/src/components/Overview/Overview.jsx:28]: Yeah, I have noticed that with the style guide. I feel like it makes things more cluttered and less legible but I'm fine with adhering to that style. I'll push a change

Comment (zman811) [reply client/src/components/Overview/Overview.jsx:28]: I think it makes it easier to see its a func at a glance but I do agree that its more cluttered

Comment (slargman) [reply client/src/components/Overview/Overview.jsx]: yeah thinking about it, I agree with you. Helps eliminate the symbol soup that can occur.

**Comment by slargman** [comment]
Thanks for the review Zach! I'll go ahead and merge.

**Pull request merged**